app/services/provider_interface/active_application_statuses_by_provider.rb

```diff
@@ -0,0 +1,42 @@
+module ProviderInterface
+  class ActiveApplicationStatusesByProvider
+    attr_reader :provider
+
+    def initialize(provider)
+      @provider = provider
+    end
+
+    def call
+      grouped_course_data.map do |course_data|
+        courses = course_data.last
+        {
+          header: courses.first.name_and_code,
+          subheader: provider_name(courses.first),
+          values: [status_count(courses, :awaiting_provider_decision),
+                   status_count(courses, :interviewing),
+                   status_count(courses, :offer),
+                   status_count(courses, :pending_conditions),
+                   status_count(courses, :recruited)],
+        }
+      end
+    end
+
+  private
+
+    def grouped_course_data
+      @course_data ||= GetApplicationProgressDataByCourse.new(provider: provider).call.group_by(&:id)
```

Comment:
Why do we group by `id` here? It seems to be disregarded on line 11 in the `call` method.

Reply:
Group by id on line 11 will group the results by id, and status which results in a single course object that contains the id, status of application and other attributes that we need for our checks. So it will result in something like

```
Course id:1, status: 'pending', count: '10', provider_name: 'some provider'...
Course id:1, status: 'awaiting_provider_decision', count: '5', provider_name: 'some provider'...
Course id:1, status: 'pending', count: '', provider_name: 'other provider'...
Course id:1, status: 'pending', count: '10', provider_name: 'other provider'...
```

From there on we need to regroup things so we can correctly contruct each line of the report.
Mapping it to `course` i think is causing the confusion here but it means that we can use AR to construct the query and don't need to have any additional calls in order to retrieve information that we need.